Add docker-compose.yaml

msabramo opened the pull request:
that starts both vault and vault-ui, so that folks can more easily test.

```
$ docker-compose up -d
Creating vaultui_vault_1
Creating vaultui_vault-ui_1

$ docker-compose ps
       Name                     Command               State           Ports
------------------------------------------------------------------------------------
vaultui_vault-ui_1   npm run serve                    Up      0.0.0.0:8000->8000/tcp
vaultui_vault_1      docker-entrypoint.sh serve ...   Up      0.0.0.0:8200->8200/tcp
```

msabramo committed:
Add docker-compose.yaml

that starts both vault and vault-ui, so that folks can more easily test.

```
$ docker-compose up -d
Creating vaultui_vault_1
Creating vaultui_vault-ui_1

$ docker-compose ps
       Name                     Command               State           Ports
------------------------------------------------------------------------------------
vaultui_vault-ui_1   npm run serve                    Up      0.0.0.0:8000->8000/tcp
vaultui_vault_1      docker-entrypoint.sh serve ...   Up      0.0.0.0:8200->8200/tcp
```

djenriquez commented:
Great idea. I'm not familiar with the dockerized Vault, will running this provider users with the tokens necessary to log into the dev vault?

msabramo commented:
Yeah so when you run the dockerized Vault with no args, you get a dev server with a random token generated. The users would have to do something like `docker-compose logs -f vault` to see and copy the token -- e.g.:

```
$ docker-compose logs -f vault
Attaching to vaultui_vault_1
vault_1     | ==> Vault server configuration:
vault_1     |
vault_1     |                  Backend: inmem
vault_1     |               Listener 1: tcp (addr: "0.0.0.0:8200", cluster address: "", tls: "disabled")
vault_1     |                Log Level: info
vault_1     |                    Mlock: supported: true, enabled: false
vault_1     |                  Version: Vault v0.6.2
...
vault_1     | Unseal Key: A/9ci8Lk6kJ1UixnqFR8yyWMKN391Fx5aMBHaFw7Rj0=
vault_1     | Root Token: d1ea5160-4fca-93a6-e686-3b817bc675fb
vault_1     |
vault_1     | ==> Vault server started! Log data will stream in below:
vault_1     |
vault_1     | 2016/12/04 17:06:21.472124 [INFO ] core: successfully mounted backend: type=generic path=secret/
vault_1     | 2016/12/04 17:06:21.472136 [INFO ] core: successfully mounted backend: type=cubbyhole path=cubbyhole/
vault_1     | 2016/12/04 17:06:21.472193 [INFO ] core: successfully mounted backend: type=system path=sys/
```

I'm thinking now that we can make this easier. The easy thing to do is to set an environment variable when starting Vault the hard-codes the root token to something simple like all zeros. But maybe even better is to enable the `userpass` authentication backend and add a simple dummy username/password combo like `test/test`.

djenriquez commented:
That would be nice, but for now I think this would suffice. I think some instructions in the README for grabbing the root token should be good enough.

msabramo committed:
docker-compose.yaml: Pick up changes

by mounting the current directory to /app and running a container that
runs `webpack -w`.

This allows one to use the docker-compose environment for development.
You can change files and instantly see your results.

msabramo commented:
Hey @djenriquez, check out 098301b. It lets one use this as a development environment, as it will pick up changes and instantly reflect them.

djenriquez commented:
Looks good @msabramo, could you also update the docs to reflect the changes? I'll merge in after.

msabramo committed:
docker-compose: Set up vault with `userpass` auth

and create a `test/test` user/password so that folks can easily log in
with Vault UI.

```
$ ./run-docker-compose-dev
Creating vaultui_vault_1
Creating vaultui_webpack_1
Creating vaultui_vault-ui_1

------------- vault status -------------
Sealed: false
Key Shares: 1
Key Threshold: 1
Unseal Progress: 0
Version: 0.6.2
Cluster Name: vault-cluster-a21dffd3
Cluster ID: 6a758590-da19-8a78-37d4-6c469bf0aa88

High-Availability Enabled: false

------------- vault auth-enable userpass -------------
Successfully enabled 'userpass' at 'userpass'!

------------- vault policy-write admin /app/admin.hcl -------------
Policy 'admin' written.

------------- vault write auth/userpass/users/test password=test policies=admin -------------
Success! Data written to: auth/userpass/users/test
```

msabramo committed:
README.md: Document using Docker for dev

msabramo commented:
@djenriquez: I now set up vault with `userpass` auth and a username `test` with password `test`. And I documented how to use it.

djenriquez commented:
Thanks @msabramo
